### PR TITLE
Add luacheck to CI

### PR DESCRIPTION
**Pull request by glacambre**

### CODE REVIEW

**Comment by alerque** [comment]
You can add a `.luachechrc` to suppress the warning about the global `vim` variable it doesn't understand. The line length warning might be the other one that you don't need. All the other warnings look legit to me.

```lua
std = "lua53"
globals = {
  "vim"
}
max_line_length = false
```

**Comment by alerque** [comment]
If you want me to submit a PR against this branch with that config and the fixes for the lint errors let me know.

File: .github/workflows/luacheck.yml [hunk 1]
```diff
@@ -0,0 +1,24 @@
+name: Luacheck
+
+on:
+  push:
+    branches:
+    - master
+  pull_request:
+    branches:
+    - master
```
Comment (alerque) [inline .github/workflows/luacheck.yml:9]: Shouldn't this just be on all pull requests with no branch filter?

**Comment by alerque** [comment]
Unfortunately the Ubuntu Lua stuff is 5.1 still. If you'd like to do this lint on 5.3 you can copy [this incantation](https://github.com/sile-typesetter/casile/blob/master/.github/workflows/luacheck.yml).

**Comment by glacambre** [comment]
Thanks for taking a look and offering to help - I almost fixed everything 12 hours ago but I had to got to work before being able to complete my work and push it.

Luacheck being stuck on 5.1 is fine because most of the time Neovim is shipped with luajit, which only support lua 5.1 :).

**Comment by glacambre** [comment]
Travis disappeared from the checks. Not sure if this is because of this PR. Probably not, someone seems to have the same issue somewhere else: https://travis-ci.community/t/travis-disappeared-from-github-com-pull-request-checklist/7556

**Comment by alerque** [comment]
@glacambre Were you authed against .com or .org for Travis. It likely has nothing to do with this PR, but they've been deprecating some of their older auth and trigger systems.

**Comment by glacambre** [comment]
I have no idea. The odd thing is that LGTM also disappeared from the checks. But since this also happened for other PRs I guess this one isn't to blame.

I've been meaning to switch away from Travis for a while, I guess I'll do it now since they're forcing my hand :).

**Pull request merged**